lib/bolt/config.rb

```diff
@@ -139,9 +139,22 @@ def self.load_bolt_defaults_yaml(dir)
         )
       end
 
-      # Move data under transport-config to top-level so it can be easily merged with
-      # config from other sources.
+      # Move data under inventory-config to top-level so it can be easily merged with
+      # config from other sources. Error early if inventory-config is not a hash or
+      # has a plugin reference.
       if data.key?('inventory-config')
+        unless data['inventory-config'].is_a?(Hash)
+          raise Bolt::ValidationError,
+                "Option 'inventory-config' must be of type Hash, received #{data['inventory-config']} "\
+                "#{data['inventory-config']} (file: #{filepath})"
+        end
+
+        if data['inventory-config'].key?('_plugin')
```

Comment:
Why do we not support plugins here? I thought we did some work to allow transport config in `bolt.yaml` to support plugin references. Would this be a regression or is it just not supported at the _top_ level?

Reply:
It's just the top-level `inventory-config` key that can't be a plugin reference. The individual transports can be plugin references, though.